api/proto/agent/actions.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto3";
+
+package scheduler;
+
+enum RunnerType {
+    RUNNER_TYPE_DOCKER = 0;
+}
+
+message ExecutionContext {
+    RunnerType type = 1;
+    optional string container_image = 2;
+}
+
+message ActionRequest {
+    string action_id = 1;
+    ExecutionContext context = 2;
+    repeated string commands = 3;
+}
+
+enum ActionStatus {
+    ACTION_STATUS_TODO = 0;
+    ACTION_STATUS_PENDING = 1;
+    ACTION_STATUS_RUNNING = 2;
+    ACTION_STATUS_COMPLETED = 3;
+    ACTION_STATUS_ERROR = 4;
```

Comment:
You don't need that status. An action that completed with a non-zero exit code describes an error state.